src/pages/projectSettings/Header.tsx

```diff
@@ -1,186 +1,39 @@
+import { readOnlyTabs, WritableTabRoutes } from "./tabs/types";
 import { ProjectType } from "./tabs/utils";
 
 interface Props {
   id: string;
-  isRepo: boolean;
   projectType: ProjectType;
-  saveable: boolean;
   tab: ProjectSettingsTabRoutes;
 }
 
-export const Header: React.VFC<Props> = ({
-  id,
-  isRepo,
-  projectType,
-  saveable,
-  tab,
-}) => {
-  const { sendEvent } = useProjectSettingsAnalytics();
-  const dispatchToast = useToastContext();
-
+export const Header: React.VFC<Props> = ({ id, projectType, tab }) => {
   const { title, subtitle } = getTabTitle(tab);
-  const { getTab, saveTab } = useProjectSettingsContext();
-  const { formData, hasChanges, hasError } = getTab(tab);
-  const navigate = useNavigate();
-  const { identifier } = useParams<{ identifier: string }>();
-
-  const [defaultModalOpen, setDefaultModalOpen] = useState(false);
-
-  const [saveProjectSection] = useMutation<
-    SaveProjectSettingsForSectionMutation,
-    SaveProjectSettingsForSectionMutationVariables
-  >(SAVE_PROJECT_SETTINGS_FOR_SECTION, {
-    onCompleted({
-      saveProjectSettingsForSection: {
-        projectRef: { identifier: newIdentifier },
-      },
-    }) {
-      saveTab(tab);
-      dispatchToast.success("Successfully updated project");
-
-      if (identifier !== newIdentifier) {
-        navigate(getProjectSettingsRoute(newIdentifier, tab), {
-          replace: true,
-        });
-      }
-    },
-    onError(err) {
-      dispatchToast.error(
-        `There was an error saving the project: ${err.message}`
-      );
-    },
-    refetchQueries: ({
-      data: {
-        saveProjectSettingsForSection: {
-          projectRef: { identifier: newIdentifier },
-        },
-      },
-    }) =>
-      identifier === newIdentifier
-        ? ["ProjectSettings", "GetViewableProjectRefs"]
-        : [],
-  });
-
-  const [saveRepoSection] = useMutation<
-    SaveRepoSettingsForSectionMutation,
-    SaveRepoSettingsForSectionMutationVariables
-  >(SAVE_REPO_SETTINGS_FOR_SECTION, {
-    onCompleted() {
-      saveTab(tab);
-      dispatchToast.success("Successfully updated repo");
-    },
-    onError(err) {
-      dispatchToast.error(`There was an error saving the repo: ${err.message}`);
-    },
-    refetchQueries: ["RepoSettings", "GetViewableProjectRefs"],
-  });
-
-  const onClick = () => {
-    const newData = formToGqlMap[tab](formData, id);
-    const save = (update, section) =>
-      isRepo
-        ? saveRepoSection({
-            variables: {
-              section,
-              repoSettings: update,
-            },
-          })
-        : saveProjectSection({
-            variables: {
-              section,
-              projectSettings: update,
-            },
-          });
-
-    const section = mapRouteToSection[tab];
-    save(newData, section);
-    sendEvent({ section, name: isRepo ? "Save repo" : "Save project" });
-  };
+  const saveable = !(readOnlyTabs as ReadonlyArray<string>).includes(tab);
```

Comment:
Since `readOnlyTabs` is defined with a const assertion, it needs to be cast to a ReadonlyArray to use `.includes()` on it (see [here](https://stackoverflow.com/a/56745484))